pkg/i2gw/ingress2gateway.go

```diff
@@ -47,8 +47,9 @@ func ToGatewayAPIResources(ctx context.Context, namespace string, inputFile stri
 	}
 
 	providerByName, err := constructProviders(&ProviderConf{
-		Client:    clusterClient,
-		Namespace: namespace,
+		Client:           clusterClient,
+		Namespace:        namespace,
+		ProviderSpecific: providerSpecificConf,
```

Comment:
I'd already commented on this [here](https://github.com/kubernetes-sigs/ingress2gateway/commit/12dd6a09cceb21f537a2c134e01d64f012665630#r141876286)

I still think we need to change that, not sure I understood your point of view from the answer there. Maybe providerSpecificFlags? ExtraFlags?